src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -183,17 +191,46 @@ public static bool UpdateInstallerNodes(InstallerManifest installerManifest, IEn
             // Update previous installers with parsed data from downloaded packages
             foreach (var newInstaller in newInstallers)
             {
-                // Find a match for the current installer in the list of installers remaining to be updated.
-                var matchingExistingInstaller = existingInstallers.SingleOrDefault(i =>
-                    (i.InstallerType ?? installerManifest.InstallerType) == newInstaller.InstallerType &&
+                var archGuess = GetArchFromUrl(newInstaller.InstallerUrl);
```

Comment:
Now that we have this cool logic, I wonder if we should use it in the new command too? For example, this currently detects the wrong arch type:
```
new https://download.videolan.org/vlc/3.0.15/win64/vlc-3.0.15-win64.exe https://download.videolan.org/vlc/3.0.15/win32/vlc-3.0.15-win32.exe
```
What about adding the override after PackageParser:337 too? (And log a warning if we detect a mismatch)

Reply:
> > Check Firefox to see how we handle apps that differ only by locale
> 
> @ryfu-msft, how'd this go?

As expected, installer nodes that differed only by locale had a similar behavior to installer nodes that only differed by scope. We will need to investigate the heuristics to see if we can catch these cases.

Reply:
Added the cool logic to the new command. New command now shows the warning messages if a arch mismatch is detected prior to prompting for required fields.